test/test_output_engine.py

```diff
@@ -13,53 +13,60 @@
 class TestOutputEngine(unittest.TestCase):
     """ Test the OutputEngine class functions """
 
-    MOCK_MODULES = {
+    MOCK_OUTPUT = {
         "modulename0": {
-            "1.0": {"CVE-1234-1234": "MEDIUM", "CVE-1234-9876": "LOW",},
-            "2.8.6": {"CVE-1234-1111": "LOW",},
+            "1.0": {"CVE-1234-1234": "MEDIUM", "CVE-1234-9876": "LOW"},
+            "2.8.6": {"CVE-1234-1111": "LOW"},
         },
-        "modulename1": {"3.2.1.0": {"CVE-1234-5678": "HIGH",}},
+        "modulename1": {"3.2.1.0": {"CVE-1234-5678": "HIGH"}},
     }
-    FORMATTED_MODULES = {
-        "modulename0": {
-            "1.0": [
-                {"cve": "CVE-1234-1234", "severity": "MEDIUM"},
-                {"cve": "CVE-1234-9876", "severity": "LOW"},
-            ],
-            "2.8.6": [{"cve": "CVE-1234-1111", "severity": "LOW"},],
+    FORMATTED_OUTPUT = [
+        {
+            "package": "modulename0",
+            "version": "1.0",
+            "cve_number": "CVE-1234-1234",
+            "severity": "MEDIUM",
         },
-        "modulename1": {"3.2.1.0": [{"cve": "CVE-1234-5678", "severity": "HIGH"},]},
-    }
+        {
+            "package": "modulename0",
+            "version": "1.0",
+            "cve_number": "CVE-1234-9876",
+            "severity": "LOW",
+        },
+        {
+            "package": "modulename0",
+            "version": "2.8.6",
+            "cve_number": "CVE-1234-1111",
+            "severity": "LOW",
+        },
+        {
+            "package": "modulename1",
+            "version": "3.2.1.0",
+            "cve_number": "CVE-1234-5678",
+            "severity": "HIGH",
+        },
+    ]
 
     def setUp(self):
-        self.output_engine = OutputEngine(modules=self.MOCK_MODULES)
-        self.mock_file = io.StringIO()
+        self.output_engine = OutputEngine(modules=self.MOCK_OUTPUT)
+        self.mock_file = tempfile.TemporaryFile("w+")
```

Comment:
We need to make sure we cleanup this temporary file, let's add a `tearDown` method.

Reply:
okay I will use NamedTemporaryFile() which  deletes file when we close it.

Reply:
I have fixed it.